.github/workflows/build.yml

```diff
@@ -36,6 +36,37 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+      # Check whether gradle.properties version has been released already
+      # This prevents accidently publish same version
+      # Only check binary for arithmetic as all published together and arithmetic is the first
+      - name: Check release published
+        run: |
+          # Skip if version contain SNAPSHOT
+          grep "^version" gradle.properties | grep -q SNAPSHOT || {
```

Comment:
how about using a negated regex to positively identify a version rather than checking for SNAPSHOT, like 
`grep "^version" gradle.properties | grep -qvE '^version=[0-9]+\.[0-9]+\.[0-9]+$' ||`

I will say that using a conditional if/else statement is more readable than short circuit evaluation though

Reply:
Fixes added